lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -159,37 +155,40 @@ def get_event_type(event)
 
     # Rescue retryable errors during bulk submission
     def safe_bulk(es_actions,actions)
-      @client.bulk(es_actions)
-    rescue Manticore::SocketException, Manticore::SocketTimeout => e
-      # If we can't even connect to the server let's just print out the URL (:hosts is actually a URL)
-      # and let the user sort it out from there
-      @logger.error(
-        "Attempted to send a bulk request to Elasticsearch configured at '#{@client.client_options[:hosts]}',"+
-          " but Elasticsearch appears to be unreachable or down!",
-        :client_config => @client.client_options,
-        :error_message => e.message,
-        :class => e.class.name
-      )
-      @logger.debug("Failed actions for last bad bulk request!", :actions => actions)
-
-      # We retry until there are no errors! Errors should all go to the retry queue
-      sleep @retry_max_interval
-      retry unless @stopping.true?
-    rescue => e
-      # For all other errors print out full connection issues
-      @logger.error(
-        "Attempted to send a bulk request to Elasticsearch configured at '#{@client.client_options[:hosts]}'," +
-          " but an error occurred and it failed! Are you sure you can reach elasticsearch from this machine using " +
-          "the configuration provided?",
-        :client_config => @client.client_options,
-        :error_message => e.message,
-        :error_class => e.class.name,
-        :backtrace => e.backtrace
-      )
-
-      @logger.debug("Failed actions for last bad bulk request!", :actions => actions)
-
-      raise e
+      sleep_interval = @retry_initial_interval
+      begin
+        url, response = @client.bulk(es_actions)
+        response
+      rescue Manticore::SocketException, Manticore::SocketTimeout => e
```

Comment:
in https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/390/files#diff-885c6dfd0ed36754f38c5c925b275036R53 we define a list of exceptions known to mean unreachable host, and this list is then used in https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/390/files#diff-ba322aad25863c7bb2706a30288a4416R208.

This means SocketException will never reach this, while HostUnreachableError might